src/TBuffer.cpp

```diff
@@ -779,8 +811,8 @@ TBuffer::TBuffer(Host* pH)
 
     // Validate the encoding tables in case there has been an edit which breaks
     // things:
-    for (auto table : csmEncodingTable) {
-        Q_ASSERT_X(table.second.size() == 128, "TBuffer", "Mis-sized encoding look-up table.");
+    for (auto table : csmEncodingsMap) {
+        Q_ASSERT_X(table.size() == 128, "TBuffer", "Mis-sized encoding look-up table.");
```

Comment:
We should change this to the compile-time `static_assert()` available in C++11: faster feedback.

Reply:
Does it not get MACROed out to that _when it can be_ by the Qt Libaries?

Reply:
What do you mean by when it can be - we're already using C++11, there's no way it can't be.

I don't think it is, Qt documentation says nothing about it. I think this is a runtime check.

Reply:
I see what you mean with the difference between a compile and run-time checks and yes it would be better to check it once during compilation than every time a `TBuffer` is constructed (especially as the data is a globally shared single instance).

I am just not quite sure of the best way to reliable ensure that each table in the set is exactly 128 elements because if it is not then a source code edit has messed something up and it will cause _undefined behaviour_ if the table is shorter and probably data corruption if it is longer...

Reply:
I'm not sure either, but this would be a learning opportunity. I think a combination of a constexpr function + static_assert would do it.

Reply:
I've had a look, I think we need C++14 at minimum to do this. Either way, this is not something you have to address for this PR for clarity's sake.